devito/arch/archinfo.py

```diff
@@ -41,7 +41,7 @@
            'PVC', 'INTELGPUMAX', 'MAX1100', 'MAX1550']
 
 
-@memoized_func
+@cache
```

Comment:
With `memoized_func` I know, IIRC, that the memoization cannot be evicted for the entire lifetime of the program. Is it still the same here? is this specific cache unbounded? otherwise, we may need an unbounded cache that we only surgically use for selected "system" functions, otherwise we would introduce a change in functionality